compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -62,13 +60,15 @@ class TryInfo(val onExit: IrExpression) : ExpressionInfo() {
     val gaps = mutableListOf<Pair<Label, Label>>()
 }
 
-class BlockInfo private constructor(val parent: BlockInfo?) {
-    val variables = mutableListOf<VariableInfo>()
-    val infos = Stack<ExpressionInfo>()
+class ReturnableBlockInfo(
+    val returnLabel: Label,
+    val returnSymbol: IrSymbol,
+    val returnTemporary: Int? = null
+) : ExpressionInfo()
 
-    fun create() = BlockInfo(this).apply {
-        this@apply.infos.addAll(this@BlockInfo.infos)
-    }
+class BlockInfo(val parent: BlockInfo? = null) {
+    val variables = mutableListOf<VariableInfo>()
+    val infos: Stack<ExpressionInfo> = parent?.infos ?: Stack()
```

Comment:
Shouldn't 'parent?.infos' be copied to another stack?

Reply:
Doesn't matter, wirhBlock/handleBlock guarantee that scopes are exited correctly before destruction of BlockInfo.

Reply:
@pyos Than it's best to hide this field: `private` (that will require to make some of inline function 'internal')